explorer/ast/statement.cpp

```diff
@@ -15,39 +15,86 @@ using llvm::cast;
 
 Statement::~Statement() = default;
 
-void Statement::PrintDepth(int depth, llvm::raw_ostream& out) const {
-  if (depth == 0) {
-    out << " ... ";
-    return;
+void Statement::PrintID(llvm::raw_ostream& out) const {
+  switch (kind()) {
+    case StatementKind::Match: {
+      const auto& match = cast<Match>(*this);
+      out << "match (" << match.expression() << ") { ... }";
+      break;
+    }
+    case StatementKind::While: {
+      const auto& while_stmt = cast<While>(*this);
+      out << "while (" << while_stmt.condition() << ") { ... }";
+      break;
+    }
+    case StatementKind::For: {
+      const auto& for_stmt = cast<For>(*this);
+      out << "for (" << for_stmt.variable_declaration() << " in "
+          << for_stmt.loop_target() << ") { ... }";
+      break;
+    }
+
+    case StatementKind::If: {
+      const auto& if_stmt = cast<If>(*this);
+      out << "if (" << if_stmt.condition() << ") { ... }";
+      if (if_stmt.else_block()) {
+        out << " else { ... }";
+      }
+      break;
+    }
+    case StatementKind::Block: {
+      const auto& block = cast<Block>(*this);
+      const auto statements = block.statements();
+      out << "{";
+      for (const auto* statement : statements) {
+        statement->Print(out);
+      }
+      out << "}";
+      break;
+    }
+    case StatementKind::Break:
+    case StatementKind::Continue:
+    case StatementKind::VariableDefinition:
+    case StatementKind::ExpressionStatement:
+    case StatementKind::Assign:
+    case StatementKind::IncrementDecrement:
+    case StatementKind::ReturnVar:
+    case StatementKind::ReturnExpression: {
+      Print(out);
+    }
+  }
+}
+
+void Statement::PrintIndent(int indent_num_spaces,
+                            llvm::raw_ostream& out) const {
+  if (kind() != StatementKind::Block) {
+    out.indent(indent_num_spaces);
   }
+
   switch (kind()) {
     case StatementKind::Match: {
       const auto& match = cast<Match>(*this);
-      out << "match (" << match.expression() << ") {";
-      if (depth < 0 || depth > 1) {
+      out << "match (" << match.expression() << ") {\n";
+      for (const auto& clause : match.clauses()) {
+        out.indent(indent_num_spaces + 2)
+            << "case " << clause.pattern() << " => \n";
+        clause.statement().PrintIndent(indent_num_spaces + 2, out);
```

Comment:
```suggestion
        clause.statement().PrintIndent(indent_num_spaces, out);
```

This is actually always a block, so probably shouldn't increase indent.

For reference, from parser.ypp:
```
clause:
  CASE pattern DOUBLE_ARROW block
    { $$ = Match::Clause($[pattern], $[block]); }
| DEFAULT DOUBLE_ARROW block
    {
      $$ = Match::Clause(arena->New<BindingPattern>(
                             context.source_loc(), std::string(AnonymousName),
                             arena->New<AutoPattern>(context.source_loc()),
                             ExpressionCategory::Value),
                         $[block]);
    }
;
```

Reply:
Doing this is leading to this output"
```
    match (self.element) {
      case OptionalElement(T).None =>
    {
      return false;
    }
    }
```

Note: for clause the indent is `indent_num_spaces + 2` as it is nested inside the match block.

Reply:
Ah, thanks, I'd missed it matching the clause indent.